addons/web/static/tests/fields/relational_fields/field_many2one_tests.js

```diff
@@ -1115,6 +1115,41 @@ QUnit.module('fields', {}, function () {
             form.destroy();
         });
 
+        QUnit.test('form: quick create for field that returns false after name_create call', async function (assert) {
+            assert.expect(3);
+
+            var prom = testUtils.makeTestPromise();
+            var form = await createView({
+                View: FormView,
+                model: 'partner',
+                data: this.data,
+                arch: '<form>' +
+                    '<field name="trululu"/>' +
+                    '</form>',
+                mockRPC: function (route, args) {
+                    var result = this._super.apply(this, arguments);
+                    if (args.method === 'name_create') {
+                        assert.step('name_create');
+                        // Resolve the name_create call to false. This is possible if
+                        // _rec_name for the model of the field is unassigned.
+                        return prom.then(_.constant(false));
```

Comment:
`return false;` should do the trick

Reply:
performRpc doesn't work if this doesn't return a promise.

Reply:
`return Promise.resolve(false);` then :-)

Reply:
yes, that's what I did. I also followed your other suggestions. The code is now slimmer and direct to the point. :)